languages/tree-sitter-stack-graphs-typescript/rust/npm_package.rs

```diff
@@ -0,0 +1,77 @@
+// -*- coding: utf-8 -*-
+// ------------------------------------------------------------------------------------------------
+// Copyright © 2022, stack-graphs authors.
+// Licensed under either of Apache License, Version 2.0, or MIT license, at your option.
+// Please see the LICENSE-APACHE or LICENSE-MIT files in this distribution for license details.
+// ------------------------------------------------------------------------------------------------
+
+use serde::Deserialize;
+use stack_graphs::graph::NodeID;
+use std::collections::HashMap;
+use std::path::Path;
+
+use stack_graphs::arena::Handle;
+use stack_graphs::graph::File;
+use stack_graphs::graph::StackGraph;
+use tree_sitter_stack_graphs::FileAnalyzer;
+use tree_sitter_stack_graphs::LoadError;
+
+use crate::util::*;
+
+pub struct NpmPackageAnalyzer {}
+
+impl FileAnalyzer for NpmPackageAnalyzer {
+    fn build_stack_graph_into<'a>(
+        &self,
+        graph: &mut StackGraph,
+        file: Handle<File>,
+        _path: &Path,
+        source: &str,
+        _all_paths: &mut dyn Iterator<Item = &'a Path>,
+        globals: &HashMap<String, String>,
+        _cancellation_flag: &dyn tree_sitter_stack_graphs::CancellationFlag,
+    ) -> Result<(), tree_sitter_stack_graphs::LoadError> {
+        // read globals
+        let proj_name = globals
+            .get(crate::PROJECT_NAME_VAR)
+            .map(String::as_str)
+            .unwrap_or("");
```

Comment:
I consider it optional, because of the inconvenience of making it required. If it was required, users would have to specify it to every CLI invocation, all the tests would need to provide values, etc. Some of these we might be able to fix with implicit defaults, but I thought this was the easiest solution.

Note that this corresponds to https://github.com/github/stack-graphs/blob/analyze-tsconfig/languages/tree-sitter-stack-graphs-typescript/src/stack-graphs.tsg#L24 as well, so if we'd make a different decision here, we probably want to change that as well.